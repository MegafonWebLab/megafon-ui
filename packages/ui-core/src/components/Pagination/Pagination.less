@import (reference) '~styles/base.less';

@b: e('.mfui-pagination');

@{b} {
    display: flex;
}
