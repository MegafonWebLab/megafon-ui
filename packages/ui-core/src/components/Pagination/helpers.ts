const getRange = (from: number, to: number, step = 1): number[] => {
    let i = from;
    const range: number[] = [];

    while (i <= to) {
        range.push(i);
        i += step;
    }

    return range;
};

export default getRange;
