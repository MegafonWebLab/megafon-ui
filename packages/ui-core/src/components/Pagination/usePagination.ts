/* eslint-disable no-magic-numbers */
/* eslint-disable import/no-unresolved */
/* eslint-disable import/extensions */
import throttleTime from 'constants/throttleTime';
import throttle from 'lodash.throttle';
import React from 'react';
import getRange from './helpers';

export const Button = {
    HIDDEN: 'HIDDEN',
    FIRST: 1,
};

const NeighbourCount = {
    MOBILE: 0,
    MAIN: 1,
};

const BUTTON_RATIO = 5;
export const MOBILE_RESOLUTION = 360;

type GetItemsParamsType = {
    totalPages: number;
    activePage: number;
    maxButtonsCount: number;
    neighbourCount: number;
};

const getItems = ({ totalPages, activePage, maxButtonsCount, neighbourCount }: GetItemsParamsType) => {
    const isMoreThenMaxBtnsCount = totalPages > maxButtonsCount;
    const lastPage = totalPages;

    if (!isMoreThenMaxBtnsCount) {
        return getRange(Button.FIRST, lastPage);
    }

    const hasLeftHiddenBtns = activePage > Math.ceil(maxButtonsCount / 2);
    const hasRightHiddenBtns = lastPage > activePage + neighbourCount + 2;

    switch (true) {
        case !hasLeftHiddenBtns && hasRightHiddenBtns: {
            const range = getRange(Button.FIRST, maxButtonsCount - 2);

            return [...range, Button.HIDDEN, lastPage];
        }

        case hasLeftHiddenBtns && !hasRightHiddenBtns: {
            const range = getRange(totalPages - (2 + 2 * neighbourCount), totalPages);

            return [Button.FIRST, Button.HIDDEN, ...range];
        }

        default: {
            const range = getRange(activePage - neighbourCount, activePage + neighbourCount);

            return [Button.FIRST, Button.HIDDEN, ...range, Button.HIDDEN, lastPage];
        }
    }
};

const usePagination = (totalPages: number, activePage: number): (string | number)[] => {
    const [neighbourCount, setNeighbourCount] = React.useState(NeighbourCount.MAIN);
    const maxButtonsCount = 2 * neighbourCount + BUTTON_RATIO;
    const paginationItems = getItems({
        totalPages,
        activePage,
        maxButtonsCount,
        neighbourCount,
    });

    React.useEffect(() => {
        const handleResize = () => {
            const isMobile = window.innerWidth <= MOBILE_RESOLUTION;
            const value = isMobile ? NeighbourCount.MOBILE : NeighbourCount.MAIN;

            setNeighbourCount(value);
        };

        handleResize();

        const handleResizeThrottled = throttle(handleResize, throttleTime.resize);

        window.addEventListener('resize', handleResizeThrottled);

        return () => {
            window.removeEventListener('resize', handleResizeThrottled);
        };
    }, []);

    return paginationItems;
};

export default usePagination;
