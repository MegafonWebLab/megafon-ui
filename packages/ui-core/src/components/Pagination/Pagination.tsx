import React from 'react';
import { cnCreate, filterDataAttrs } from '@megafon/ui-helpers';
import PropTypes from 'prop-types';
import './Pagination.less';
import PaginationButtons from './components/PaginationButtons/PaginationButtons';
import PaginationNavigation from './components/PaginationNavigation/PaginationNavigation';
import usePagination, { Button } from './usePagination';

interface IPaginationProps {
    /** Дополнительные data атрибуты к внутренним элементам */
    dataAttrs?: {
        root?: Record<string, string>;
        prev?: Record<string, string>;
        next?: Record<string, string>;
        button?: Record<string, string>;
    };
    /** Общее количество страниц */
    totalPages: number;
    /** Номер текущей страницы */
    activePage: number;
    /** Тема компонента */
    theme?: 'default' | 'light';
    /** Обработчик изменения активной страницы */
    onChange: (value: number) => void;
}

const cn = cnCreate('mfui-pagination');
const Pagination: React.FC<IPaginationProps> = ({ totalPages, activePage, dataAttrs, theme = 'default', onChange }) => {
    const paginationItems: React.ReactText[] = usePagination(totalPages, activePage);

    const handleBackClick = React.useCallback(() => {
        onChange(activePage - 1);
    }, [activePage, onChange]);

    const handleNextClick = React.useCallback(() => {
        onChange(activePage + 1);
    }, [activePage, onChange]);

    const handlePageButtonClick = React.useCallback(
        value => {
            onChange(value);
        },
        [onChange],
    );

    return (
        <div {...filterDataAttrs(dataAttrs?.root)} className={cn()}>
            <PaginationNavigation
                dataAttrs={{ root: dataAttrs?.prev }}
                direction="left"
                disabled={activePage === Button.FIRST}
                onClick={handleBackClick}
                className={cn('button')}
                theme={theme}
            />
            <PaginationButtons
                dataAttrs={{ root: dataAttrs?.button }}
                items={paginationItems}
                activeButton={activePage}
                hiddenButton={Button.HIDDEN}
                theme={theme}
                onClick={handlePageButtonClick}
            />
            <PaginationNavigation
                dataAttrs={{ root: dataAttrs?.next }}
                disabled={activePage === totalPages}
                onClick={handleNextClick}
                className={cn('button')}
                theme={theme}
            />
        </div>
    );
};

Pagination.propTypes = {
    totalPages: PropTypes.number.isRequired,
    activePage: PropTypes.number.isRequired,
    theme: PropTypes.oneOf(['default', 'light']),
    dataAttrs: PropTypes.shape({
        root: PropTypes.objectOf(PropTypes.string.isRequired),
        prev: PropTypes.objectOf(PropTypes.string.isRequired),
        next: PropTypes.objectOf(PropTypes.string.isRequired),
        button: PropTypes.objectOf(PropTypes.string.isRequired),
    }),
    onChange: PropTypes.func.isRequired,
};

export default Pagination;
