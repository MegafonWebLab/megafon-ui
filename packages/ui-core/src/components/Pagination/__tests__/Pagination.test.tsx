import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Pagination from '../Pagination';

describe('Pagination', () => {
    it('should render', () => {
        const { container } = render(<Pagination totalPages={3} activePage={2} onChange={jest.fn()} />);

        expect(container).toMatchSnapshot();
    });

    it('should disable left navigation button if first page button is active', () => {
        const { getByTestId } = render(
            <Pagination
                totalPages={3}
                activePage={1}
                onChange={jest.fn()}
                dataAttrs={{ prev: { 'data-testid': 'prev' } }}
            />,
        );

        expect(getByTestId('prev')).toBeDisabled();
    });

    it('should disable right navigation button if last page button is active', () => {
        const { getByTestId } = render(
            <Pagination
                totalPages={3}
                activePage={3}
                onChange={jest.fn()}
                dataAttrs={{ next: { 'data-testid': 'next' } }}
            />,
        );

        expect(getByTestId('next')).toBeDisabled();
    });

    it('should render component with one hidden buttons of left side', () => {
        const { getAllByTestId } = render(<Pagination totalPages={10} activePage={1} onChange={jest.fn()} />);

        const hiddenButtons = getAllByTestId('hiddenButton');

        expect(hiddenButtons.length).toEqual(1);
        expect(hiddenButtons[0]).toBeDisabled();
    });

    it('should render component with one hidden buttons of right side', () => {
        const { getAllByTestId } = render(<Pagination totalPages={10} activePage={8} onChange={jest.fn()} />);

        const hiddenButtons = getAllByTestId('hiddenButton');

        expect(hiddenButtons.length).toEqual(1);
        expect(hiddenButtons[0]).toBeDisabled();
    });

    it('should render component with two hidden buttons', () => {
        const { getAllByTestId } = render(<Pagination totalPages={10} activePage={5} onChange={jest.fn()} />);

        const hiddenButtons = getAllByTestId('hiddenButton');

        expect(hiddenButtons.length).toEqual(2);
        expect(hiddenButtons[0]).toBeDisabled();
        expect(hiddenButtons[1]).toBeDisabled();
    });

    describe('render on mobile resolution', () => {
        type LocalWindowType = Omit<Window, 'innerWidth'> & {
            innerWidth: number;
        };

        const localWindow = window as LocalWindowType;
        const windowInnerWidth = window.innerWidth;

        beforeAll(() => {
            localWindow.innerWidth = 320;
        });

        afterAll(() => {
            localWindow.innerWidth = windowInnerWidth;
        });

        it('should render component with right hidden buttons', () => {
            const { getAllByTestId } = render(<Pagination totalPages={10} activePage={1} onChange={jest.fn()} />);

            const hiddenButtons = getAllByTestId('hiddenButton');

            expect(hiddenButtons.length).toEqual(1);
            expect(hiddenButtons[0]).toBeDisabled();
        });

        it('should render component with left hidden buttons', () => {
            const { getAllByTestId } = render(<Pagination totalPages={10} activePage={8} onChange={jest.fn()} />);

            const hiddenButtons = getAllByTestId('hiddenButton');

            expect(hiddenButtons.length).toEqual(1);
            expect(hiddenButtons[0]).toBeDisabled();
        });

        it('should render component with left and right hidden buttons', () => {
            const { getAllByTestId } = render(<Pagination totalPages={10} activePage={5} onChange={jest.fn()} />);

            const hiddenButtons = getAllByTestId('hiddenButton');

            expect(hiddenButtons.length).toEqual(2);
            expect(hiddenButtons[0]).toBeDisabled();
            expect(hiddenButtons[1]).toBeDisabled();
        });
    });

    describe('should call onChange', () => {
        const onChangeMock = jest.fn();

        afterEach(() => {
            jest.clearAllMocks();
        });

        it('after click on back button', () => {
            const activePage = 2;
            const { getByTestId } = render(
                <Pagination
                    totalPages={3}
                    activePage={activePage}
                    onChange={onChangeMock}
                    dataAttrs={{ prev: { 'data-testid': 'prev' } }}
                />,
            );

            fireEvent.click(getByTestId('prev'));

            expect(onChangeMock).toHaveBeenCalledWith(activePage - 1);
        });

        it('after click on next button', () => {
            const activePage = 2;
            const { getByTestId } = render(
                <Pagination
                    totalPages={3}
                    activePage={activePage}
                    onChange={onChangeMock}
                    dataAttrs={{ next: { 'data-testid': 'next' } }}
                />,
            );

            fireEvent.click(getByTestId('next'));

            expect(onChangeMock).toHaveBeenCalledWith(activePage + 1);
        });

        it('after click on page button', () => {
            const { getByTestId } = render(
                <Pagination
                    totalPages={3}
                    activePage={1}
                    onChange={onChangeMock}
                    dataAttrs={{ button: { 'data-testid': 'button' } }}
                />,
            );

            fireEvent.click(getByTestId('button[1]'));
            expect(onChangeMock).toHaveBeenCalledWith(1);
        });
    });
});
