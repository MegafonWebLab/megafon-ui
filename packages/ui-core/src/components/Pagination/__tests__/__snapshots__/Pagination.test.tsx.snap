// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render on mobile resolution should render component with left and right hidden buttons 1`] = `
<Pagination
  activePage={5}
  onChange={[MockFunction]}
  totalPages={10}
>
  <div
    className="mfui-pagination"
  >
    <PaginationNavigation
      className="mfui-pagination__button"
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      direction="left"
      disabled={false}
      onClick={[Function]}
      theme="default"
    >
      <PaginationButton
        className="mfui-pagination-navigation mfui-pagination-navigation_direction_left mfui-pagination__button"
        dataAttrs={
          Object {
            "root": undefined,
          }
        }
        disabled={false}
        onClick={[Function]}
        theme="default"
      >
        <button
          className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-navigation mfui-pagination-navigation_direction_left mfui-pagination__button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <test-file-stub
            className="mfui-pagination-navigation__icon"
          />
        </button>
      </PaginationButton>
    </PaginationNavigation>
    <PaginationButtons
      activeButton={5}
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      hiddenButton="HIDDEN"
      items={
        Array [
          1,
          "HIDDEN",
          5,
          "HIDDEN",
          10,
        ]
      }
      onClick={[Function]}
      theme="default"
    >
      <div
        className="mfui-pagination-buttons"
      >
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="0"
          onClick={[Function]}
          theme="default"
          value={1}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          disabled={true}
          key="1"
          theme="default"
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <test-file-stub />
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={true}
          key="2"
          onClick={[Function]}
          theme="default"
          value={5}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_active mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            5
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          disabled={true}
          key="3"
          theme="default"
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <test-file-stub />
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="4"
          onClick={[Function]}
          theme="default"
          value={10}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            10
          </button>
        </PaginationButton>
      </div>
    </PaginationButtons>
    <PaginationNavigation
      className="mfui-pagination__button"
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      disabled={false}
      onClick={[Function]}
      theme="default"
    >
      <PaginationButton
        className="mfui-pagination-navigation mfui-pagination-navigation_direction_right mfui-pagination__button"
        dataAttrs={
          Object {
            "root": undefined,
          }
        }
        disabled={false}
        onClick={[Function]}
        theme="default"
      >
        <button
          className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-navigation mfui-pagination-navigation_direction_right mfui-pagination__button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <test-file-stub
            className="mfui-pagination-navigation__icon"
          />
        </button>
      </PaginationButton>
    </PaginationNavigation>
  </div>
</Pagination>
`;

exports[`Pagination render on mobile resolution should render component with left hidden buttons 1`] = `
<Pagination
  activePage={8}
  onChange={[MockFunction]}
  totalPages={10}
>
  <div
    className="mfui-pagination"
  >
    <PaginationNavigation
      className="mfui-pagination__button"
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      direction="left"
      disabled={false}
      onClick={[Function]}
      theme="default"
    >
      <PaginationButton
        className="mfui-pagination-navigation mfui-pagination-navigation_direction_left mfui-pagination__button"
        dataAttrs={
          Object {
            "root": undefined,
          }
        }
        disabled={false}
        onClick={[Function]}
        theme="default"
      >
        <button
          className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-navigation mfui-pagination-navigation_direction_left mfui-pagination__button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <test-file-stub
            className="mfui-pagination-navigation__icon"
          />
        </button>
      </PaginationButton>
    </PaginationNavigation>
    <PaginationButtons
      activeButton={8}
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      hiddenButton="HIDDEN"
      items={
        Array [
          1,
          "HIDDEN",
          8,
          9,
          10,
        ]
      }
      onClick={[Function]}
      theme="default"
    >
      <div
        className="mfui-pagination-buttons"
      >
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="0"
          onClick={[Function]}
          theme="default"
          value={1}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          disabled={true}
          key="1"
          theme="default"
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <test-file-stub />
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={true}
          key="2"
          onClick={[Function]}
          theme="default"
          value={8}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_active mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            8
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="3"
          onClick={[Function]}
          theme="default"
          value={9}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            9
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="4"
          onClick={[Function]}
          theme="default"
          value={10}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            10
          </button>
        </PaginationButton>
      </div>
    </PaginationButtons>
    <PaginationNavigation
      className="mfui-pagination__button"
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      disabled={false}
      onClick={[Function]}
      theme="default"
    >
      <PaginationButton
        className="mfui-pagination-navigation mfui-pagination-navigation_direction_right mfui-pagination__button"
        dataAttrs={
          Object {
            "root": undefined,
          }
        }
        disabled={false}
        onClick={[Function]}
        theme="default"
      >
        <button
          className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-navigation mfui-pagination-navigation_direction_right mfui-pagination__button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <test-file-stub
            className="mfui-pagination-navigation__icon"
          />
        </button>
      </PaginationButton>
    </PaginationNavigation>
  </div>
</Pagination>
`;

exports[`Pagination render on mobile resolution should render component with right hidden buttons 1`] = `
<Pagination
  activePage={1}
  onChange={[MockFunction]}
  totalPages={10}
>
  <div
    className="mfui-pagination"
  >
    <PaginationNavigation
      className="mfui-pagination__button"
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      direction="left"
      disabled={true}
      onClick={[Function]}
      theme="default"
    >
      <PaginationButton
        className="mfui-pagination-navigation mfui-pagination-navigation_direction_left mfui-pagination__button"
        dataAttrs={
          Object {
            "root": undefined,
          }
        }
        disabled={true}
        onClick={[Function]}
        theme="default"
      >
        <button
          className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-navigation mfui-pagination-navigation_direction_left mfui-pagination__button"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <test-file-stub
            className="mfui-pagination-navigation__icon"
          />
        </button>
      </PaginationButton>
    </PaginationNavigation>
    <PaginationButtons
      activeButton={1}
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      hiddenButton="HIDDEN"
      items={
        Array [
          1,
          2,
          3,
          "HIDDEN",
          10,
        ]
      }
      onClick={[Function]}
      theme="default"
    >
      <div
        className="mfui-pagination-buttons"
      >
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={true}
          key="0"
          onClick={[Function]}
          theme="default"
          value={1}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_active mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="1"
          onClick={[Function]}
          theme="default"
          value={2}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            2
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="2"
          onClick={[Function]}
          theme="default"
          value={3}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          disabled={true}
          key="3"
          theme="default"
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <test-file-stub />
          </button>
        </PaginationButton>
        <PaginationButton
          className="mfui-pagination-buttons__button"
          dataAttrs={
            Object {
              "root": Object {},
            }
          }
          isActive={false}
          key="4"
          onClick={[Function]}
          theme="default"
          value={10}
        >
          <button
            className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-buttons__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            10
          </button>
        </PaginationButton>
      </div>
    </PaginationButtons>
    <PaginationNavigation
      className="mfui-pagination__button"
      dataAttrs={
        Object {
          "root": undefined,
        }
      }
      disabled={false}
      onClick={[Function]}
      theme="default"
    >
      <PaginationButton
        className="mfui-pagination-navigation mfui-pagination-navigation_direction_right mfui-pagination__button"
        dataAttrs={
          Object {
            "root": undefined,
          }
        }
        disabled={false}
        onClick={[Function]}
        theme="default"
      >
        <button
          className="mfui-pagination-button mfui-pagination-button_theme_default mfui-pagination-navigation mfui-pagination-navigation_direction_right mfui-pagination__button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <test-file-stub
            className="mfui-pagination-navigation__icon"
          />
        </button>
      </PaginationButton>
    </PaginationNavigation>
  </div>
</Pagination>
`;

exports[`Pagination should render component with left and right hidden buttons 1`] = `
<div
  className="mfui-pagination"
>
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    direction="left"
    disabled={false}
    onClick={[Function]}
    theme="default"
  />
  <PaginationButtons
    activeButton={5}
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    hiddenButton="HIDDEN"
    items={
      Array [
        1,
        "HIDDEN",
        4,
        5,
        6,
        "HIDDEN",
        10,
      ]
    }
    onClick={[Function]}
    theme="default"
  />
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    disabled={false}
    onClick={[Function]}
    theme="default"
  />
</div>
`;

exports[`Pagination should render component with left hidden buttons 1`] = `
<div
  className="mfui-pagination"
>
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    direction="left"
    disabled={false}
    onClick={[Function]}
    theme="default"
  />
  <PaginationButtons
    activeButton={8}
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    hiddenButton="HIDDEN"
    items={
      Array [
        1,
        "HIDDEN",
        6,
        7,
        8,
        9,
        10,
      ]
    }
    onClick={[Function]}
    theme="default"
  />
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    disabled={false}
    onClick={[Function]}
    theme="default"
  />
</div>
`;

exports[`Pagination should render component with number of buttons does not exceed the limit 1`] = `
<div
  className="mfui-pagination"
>
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    direction="left"
    disabled={true}
    onClick={[Function]}
    theme="default"
  />
  <PaginationButtons
    activeButton={1}
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    hiddenButton="HIDDEN"
    items={
      Array [
        1,
        2,
        3,
      ]
    }
    onClick={[Function]}
    theme="default"
  />
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    disabled={false}
    onClick={[Function]}
    theme="default"
  />
</div>
`;

exports[`Pagination should render component with right hidden buttons 1`] = `
<div
  className="mfui-pagination"
>
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    direction="left"
    disabled={true}
    onClick={[Function]}
    theme="default"
  />
  <PaginationButtons
    activeButton={1}
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    hiddenButton="HIDDEN"
    items={
      Array [
        1,
        2,
        3,
        4,
        5,
        "HIDDEN",
        10,
      ]
    }
    onClick={[Function]}
    theme="default"
  />
  <PaginationNavigation
    className="mfui-pagination__button"
    dataAttrs={
      Object {
        "root": undefined,
      }
    }
    disabled={false}
    onClick={[Function]}
    theme="default"
  />
</div>
`;
